src/dmd/expressionsem.d

```diff
@@ -13356,3 +13362,86 @@ Expression toBoolean(Expression exp, Scope* sc)
             return e;
     }
 }
+
+
+/*****************************
+ * Inserts a `toString` call on `exp` if it's a struct or class or uses `exp`
+ * as-is if it is a string. Currently errors otherwise.
+ * Params:
+ *     exp = the expression
+ *     sc = scope to evalute `exp` in
+ * Returns:
+ *     Modified expression on success, ErrorExp on error
+ */
+Expression callToString(Expression exp, Scope* sc)
```

Comment:
this is mostly a copy of the toBoolean function above. I didn't find any other code that actually calls toString in dmd yet, so this might be a first and could maybe be simplified.

Not calling .toString didn't work with the expressionsToString method for me (printed a full struct representation)